src/fs/feature/git.rs

```diff
@@ -137,23 +137,28 @@ impl GitRepo {
     /// Git statuses is only done once, and gets cached so we don’t need to
     /// re-query the entire repository the times after that.
     ///
-    /// The temporary `Processing` enum variant is used after the `git2`
-    /// repository is moved out, but before the results have been moved in!
-    /// See <https://stackoverflow.com/q/45985827/3484614>
-    fn search(&self, index: &Path, prefix_lookup: bool) -> f::Git {
-        use std::mem::replace;
-
-        let mut contents = self.contents.lock().unwrap();
-        if let GitContents::After { ref statuses } = *contents {
+    /// “Prefix lookup” means that it should report an aggregate status of all
+    /// paths starting with the given prefix (in other words, a directory).
+    fn get_status(&self, index: &Path, prefix_lookup: bool) -> f::Git {
+        {
+            let statuses = self.statuses.read().unwrap();
```

Comment:
We were planning on adding a lint for all `.unwrap`'s (just hasn't got added yet), I know it's the lock and it's likely fine but can we match on or otherwise handle these?